### PR TITLE
mgr/vol: make some improvements

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/mgr/volumes/fs/async_cloner.py [hunk 1]
```diff
@@ -221,11 +221,13 @@ def set_quota_on_clone(fs_handle, clone_volumes_pair):
 
 def do_clone(fs_client, volspec, volname, groupname, subvolname, should_cancel):
     with open_volume_lockless(fs_client, volname) as fs_handle:
-        with open_clone_subvolume_pair(fs_client, fs_handle, volspec, volname, groupname, subvolname) as clone_volumes:
-            src_path = clone_volumes[1].snapshot_data_path(clone_volumes[2])
-            dst_path = clone_volumes[0].path
+        with open_clone_subvolume_pair(fs_client, fs_handle, volspec, volname,
+                                       groupname, subvolname)
+            as (subvol0, subvol1, subvol2):
```
Comment (kotreshhr) [inline src/pybind/mgr/volumes/fs/async_cloner.py:226]: The `open_clone_subvolume_pair` returns (clone_subvol, source_subvol, src_snapname). Please name it accordingly to improve readability.

File: src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py [hunk 2]
```diff
@@ -684,16 +684,16 @@ def _get_clone_failure(self):
...
-        if not SubvolumeOpSm.is_complete_state(state) and subvolume_type == SubvolumeTypes.TYPE_CLONE:
-            subvolume_status["source"] = self._get_clone_source()
-        if SubvolumeOpSm.is_failed_state(state) and subvolume_type == SubvolumeTypes.TYPE_CLONE:
-            try:
-                subvolume_status["failure"] = self._get_clone_failure()
-            except MetadataMgrException:
-                pass
+        subvolume_status = {'state' : state.value}
+
+        if subvolume_type == SubvolumeTypes.TYPE_CLONE:
```
Comment (neesingh-rh) [inline src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py:689]: I had also come across such lines of code earlier which needs this change.Let me point you to it: https://github.com/ceph/ceph/blob/main/src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py#L311

File: src/pybind/mgr/volumes/fs/async_cloner.py (hunk 1 above)
Comment (rishabh-d-dave) [reply src/pybind/mgr/volumes/fs/async_cloner.py:226]: Sure.

File: src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py (hunk 2 above)
Comment (rishabh-d-dave) [reply src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py:689]: True. I'll modify this one too. Thanks!

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-pull-requests/125486/

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-pull-requests/125529/

**Comment by rishabh-d-dave** [comment]
Last make check failure is unrelated.

**Comment by rishabh-d-dave** [comment]
jenkins test make check

File: src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py [hunk 3]
```diff
@@ -308,13 +308,17 @@ def open(self, op_type):
...
-                                      self.subvolname))
-
-            if op_type not in self.allowed_ops_by_state(estate) and estate != SubvolumeStates.STATE_RETAINED:
-                raise VolumeException(-errno.EAGAIN, "subvolume '{0}' is not ready for operation {1}".format(
-                                      self.subvolname, op_type.value))
+            if op_type not in self.allowed_ops_by_state(estate):
+                if estate == SubvolumeStates.STATE_RETAINED:
+                    raise VolumeException(
+                        -errno.ENOENT,
+                        'subvolume "{self.subvolname}" is removed and has '
```
Comment (neesingh-rh) [inline src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py:315]: You missed writing `f` which makes it `f-string`

Comment (rishabh-d-dave) [reply src/pybind/mgr/volumes/fs/operations/versions/subvolume_v2.py:315]: Thanks!

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-11-23_13:33:32-fs:volumes-mgr-vol-misc2-testing-default-smithi/

**Comment by rishabh-d-dave** [comment]
Looks irrelavant - https://jenkins.ceph.com/job/ceph-api/65209/

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-11-23_14:37:05-fs:volumes-mgr-vol-misc2-testing-default-smithi
https://pulpito.ceph.com/rishabh-2023-11-24_05:41:33-fs:volumes-rishabh-mgr-vol-misc-testing-default-smithi
https://pulpito.ceph.com/rishabh-2023-11-24_07:31:48-fs:volumes-rishabh-mgr-vol-misc-testing-default-smithi

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-11-24_08:24:50-fs:volumes-rishabh-mgr-vol-misc-testing-default-smithi/

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-11-24_16:45:14-fs:volumes-rishabh-mgr-vol-misc-testing-default-smithi/

All tests passed.

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-11-28_12:42:13-fs:volumes-rishabh-mgr-vol-misc-testing-default-smithi/
https://pulpito.ceph.com/rishabh-2023-11-29_11:43:01-fs:volumes-mgr-vol-misc-2023nov3-testing-default-smithi/

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-11-29_12:25:49-fs:volumes-mgr-vol-misc-2023nov3-testing-default-smithi/

**Comment by rishabh-d-dave** [comment]
> https://pulpito.ceph.com/rishabh-2023-11-29_12:25:49-fs:volumes-mgr-vol-misc-2023nov3-testing-default-smithi/

Tests ran sucessfully.

**Review by vshankar — APPROVED** [review]

**Pull request merged**